holoviews/operation/element.py

```diff
@@ -758,8 +758,12 @@ def _process(self, p, element, ranges={}):
         for d1, d2 in permuted_dims:
             if d1 == d2:
                 if p.diagonal_type is not None:
-                    values = element.dimension_values(d1)
-                    el = p.diagonal_type(values, vdims=[d1])
+                    if p.diagonal_type._1d:
```

Comment:
I'm not sure I like this approach. When was it added and how much is it used?

As far as I can tell this is a class attribute and I'm coming to the conclusion there are lots of properties (such as this one) we want to query at the level of the element type.  I would rather not have an undocumented API consisting of all sorts of random class attributes all specified on the element classes.

Instead I'm thinking there could be a set of utilities (either in ``util`` or even a new file) that is all about specifying the properties of Elements in a public way. These could be a bunch of predicate functions (with docstrings to explain what the properties mean). E.g the above might be something like:

```python
if util.is_distribution1D(p.diagonal_type):
...
```

If nothing else, the issue here is that I had no idea what ``_1d`` indicates from the name!